packages/client/src/tx/types.ts

```diff
@@ -164,6 +164,75 @@ export type TxOptions<Asset> = Partial<
       }
 >
 
+export type OfflineTxExtensions<Asset> = void extends Asset
+  ? {
+      /**
+       * Nonce for the signer of the transaction.
+       */
+      nonce: number
+      /**
+       * Mortality of the transaction.
+       */
+      mortality:
+        | { mortal: false }
+        | {
+            mortal: true
+            period: number
+            startAtBlock: { height: number; hash: HexString }
+          }
+      /**
+       * Tip in fundamental units. Default: `0n`
+       */
+      tip?: bigint
+      /**
+       * Custom values for chains that have custom signed-extensions.
+       * The key of the Object should be the signed-extension name and the value
+       * is an Object that accepts 2 possible keys: one for `value`
+       * and the other one for `additionallySigned`. They both receive either
+       * the encoded value as a `Uint8Array` that should be used for the
+       * signed-extension, or the decoded value that PAPI will encode using its
+       * dynamic codecs. At least one of the 2 values must be included into the
+       * signed-extension Object.
+       */
+      customSignedExtensions?: Record<string, CustomSignedExtensionValues>
+    }
+  : {
+      /**
+       * Nonce for the signer of the transaction.
+       */
+      nonce: number
+      /**
+       * Mortality of the transaction.
+       */
+      mortality:
+        | { mortal: false }
+        | {
+            mortal: true
+            period: number
+            startAtBlock: { height: number; hash: HexString }
+          }
+      /**
+       * Asset information to pay fees, tip, etc. By default it'll use the
+       * native token of the chain.
+       */
+      asset?: Asset
```

Comment:
Not sure if this will make TS show a more inconvenient type, but if you don't want to repeat the same, you could do something like 

```ts
export type OfflineTxExtensions<Asset> = {
  nonce: number,
  mortality: { mortal: false } | "…",
  tip?: bigint,
  customSignedExtension?: "…"
} & (void extends Asset ? {} : { asset?: Asset })
```ts
```

Reply:
yep, I like it!